sources/core/Stride.Core/Settings/AppSettingsManager.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using Stride.Core.Reflection;
+
+namespace Stride.Core.Settings
+{
+    /// <summary>
+    /// Manages the loading of application settings with <see cref="IAppSettingsProvider"/>.
+    /// </summary>
+    public static class AppSettingsManager
+    {
+        /// <summary>
+        /// <see cref="AppSettings"/> instance for the application.
+        /// </summary>
+        /// <note>
+        /// Loaded with an <see cref="IAppSettingsProvider"/>. If no provider implementation is found, returns empty <see cref="AppSettings"/> instance.
+        /// </note>
+        public static AppSettings Settings
+        {
+            get
+            {
+                if (settings != null)
+                    return settings;
+
+                foreach (var assembly in AssemblyRegistry.FindAll())
+                    foreach (var type in assembly.GetTypes())
+                        if (typeof(IAppSettingsProvider).IsAssignableFrom(type))
+                        {
+                            var ctor = type.GetConstructor(new Type[0]);
```

Comment:
Maybe using `Activator.CreateInstance` is more explicit, and possibly even faster (this point doesn't matter much if called once)?

Reply:
Looked up online - `Activator.CreateInstance` can be faster for the default constructor. I used this here because iterating over all types also gave the interface itself (which I could have eliminated another way). I'll change it with the fix for the above.